chrome/content/zotero/xpcom/sync/syncAPIClient.js

```diff
@@ -30,12 +30,16 @@ if (!Zotero.Sync) {
 Zotero.Sync.APIClient = function (options) {
 	if (!options.baseURL) throw new Error("baseURL not set");
 	if (!options.apiVersion) throw new Error("apiVersion not set");
-	if (!options.apiKey) throw new Error("apiKey not set");
+	if (!(options.apiKey || (options.username && options.password))) {
+		throw new Error("neither apiKey nor (username, password) is set");
+	}
 	if (!options.caller) throw new Error("caller not set");
 	
 	this.baseURL = options.baseURL;
 	this.apiVersion = options.apiVersion;
 	this.apiKey = options.apiKey;
+	this.username = options.username;
+	this.password = options.password;
```

Comment:
Ah, right. We could drop the API key requirement in the constructor and just throw if there's no API key in `makeRequest` unless `options.noAPIKey` is passed, which would only be the case from `APIClient.getAPIKeyFromCredentials()`.